docs/upgrading-to-v0.10.md

```diff
@@ -0,0 +1,82 @@
+- A [VolumeSnapshotLocation][2] CRD to capture information about where to store your persistent volume snapshots
+- Command-line flags for the `ark server` command (run by your Ark deployment) to capture miscellaneous Ark settings
+
+When upgrading to v0.10, you'll need to transfer the configuration information that you currently have in the `Config` CRD
+into the above. We'll cover exactly how to do this below.
+
+### Reorganization of data in object storage
+
+We've made [changes to the layout of data stored in object storage][3] for simplicity and extensibility. You'll need to
+rearrange any pre-v0.10 data as part of the upgrade. We've provided a script to help with this.
+
+## Step-by-Step Upgrade Instructions
+
+1. Scale down your existing Ark deployment:
+```bash
+kubectl scale -n heptio-ark deploy/ark --replicas 0
+```
+
+2. Re-apply the `00-prereqs.yaml` file to create new CRDs:
+```bash
+kubectl apply -f examples/common/00-prereqs.yaml
+```
+
+3. Create one or more [BackupStorageLocation][1] resources based on the examples provided in the `examples/` directory for your platform, using information from
+the existing `Config` resource as necessary.
+
+4. If you're using Ark to take PV snapshots, create one or more [VolumeSnapshotLocation][2] resources based on the examples provided in the `examples/` directory for your platform, using information from
+the existing `Config` resource as necessary.
+
+5. Perform the one-time object storage migration detailed [here][3].
+
+6. In your Ark deployment YAML (see the `examples/` directory for samples), specify flags to the `ark server` command under the container's `args`:
+
+    a. The names of the `BackupStorageLocation` and `VolumeSnapshotLocation(s)` that should be used by default for backups, if none are explicitly
+    specified by the user when creating a backup. Note that by default, the Ark server looks for a `BackupStorageLocation` named `default` to
+    use.
+
+        Flag | Default Value | Description | Example
+        ---- | ------------- | ----------- | -----------------------
+        `--default-backup-storage-location` | "default" | name of the backup storage location that should be used by default for backups | aws-us-east-1-bucket
+        `--default-volume-snapshot-locations` | [none] | name of the volume snapshot location(s) that should be used by default for PV snapshots, for each PV provider | aws:us-east-1,portworx:local
+
+    **NOTE:** the values of these flags should correspond to the names of a `BackupStorageLocation` and `VolumeSnapshotLocation(s)` that you've created as
+    custom resources in the cluster, per steps (3) and (4).
+
+    b. Any non-default Ark server settings:
+
+        Flag | Default Value | Description
+        ---- | ------------- | -----------
+        `--backup-sync-period` | 1m | how often to ensure all Ark backups in object storage exist as Backup API objects in the cluster
```

Comment:
Would it make sense to provide the names of the keys from `Config` struct here, too? `restic-timeout` was `podVolumeOperationTimeout` before, which isn't really obvious.

Reply:
Maybe I'll just mention it in the description for that one?  Seems redundant for the rest.

Reply:
Yeah, that's fine. I think it was undocumented earlier, too, so I'm skeptical it was in widespread use.